{
  Throwable exception=new RuntimeException("HA!");
  when(engine.execute(anyString(),anyMapOf(String.class,Object.class))).thenThrow(exception);
  TransactionalActions actions=new TransactionalActions(mess,engine,registry,logger);
  Map<String,Object> parameters=map();
  TestResultHandler results=new TestResultHandler();
  actions.findTransaction(txId).execute(request(new Statement("My Cypher Query",parameters)),results);
  assertThat(results.results.size(),equalTo(0));
  assertThat(results.txId,equalTo(txId));
  assertThat(results.errors.get(0).getErrorCode(),equalTo(UNKNOWN_STATEMENT_ERROR));
  verify(engine).execute("My Cypher Query",parameters);
  verify(ctx).rollback();
  verify(registry).resume(txId);
  verify(registry).finish(txId);
  verifyNoMoreInteractions(mess,registry,engine,ctx);
}
